File: src/main/java/sprykerFileCreator/model/generator/ParentGeneratorInterface.java
package pav.sprykerFileCreator.model.generator;

import pav.sprykerFileCreator.model.definition.ClassDefinitionInterface;
import pav.sprykerFileCreator.model.renderer.dto.PhpClassInterface;

public interface ParentGeneratorInterface {

    PhpClassInterface getParentClass(PhpClassInterface phpClass, ClassDefinitionInterface classDefinition);

}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/services/generator/ParentGeneratorInterface.kt
package com.github.tobi812.sprykerplugin.services.generator

import com.github.tobi812.sprykerplugin.services.definitions.ClassDefinitionInterface
import com.github.tobi812.sprykerplugin.services.renderer.dto.PhpClassInterface

interface ParentGeneratorInterface {
    fun getParentClass(
        phpClass: PhpClassInterface,
        classDefinition: ClassDefinitionInterface
    ): PhpClassInterface?
}
